### PR TITLE
Update testing to new approach

### LINKED ISSUE #23: terraform destroy fails for examples/single_tunnels because of cross-project references deadlock
error output
```
       
       Error: Error reading Router: googleapi: Error 400: The router resource 'projects/gl-akopachevskyy-vpn-mgt/regions/us-central1/routers/cr-uscentral1-to-prod-vpc-tunnels' is already being used by 'projects/gl-akopachevskyy-vpn-mgt/regions/us-central1/vpnTunnels/vpn-tn-us-ce1-mgt-prd-internal-1', resourceInUseByAnotherResource
       
       
       
       Error: Error reading Router: googleapi: Error 400: The router resource 'projects/gl-akopachevskyy-vpn/regions/us-central1/routers/cr-uscentral1-to-mgt-vpc-tunnels' is already being used by 'projects/gl-akopachevskyy-vpn/regions/us-central1/vpnTunnels/vpn-tn-us-ce1-prd-mgt-internal-1', resourceInUseByAnotherResource

```
Issue discovered here #20

### PR DESCRIPTION
**Pull request by ingwarr**
Fixes https://github.com/terraform-google-modules/terraform-google-vpn/issues/19
Fixes #23

### CODE REVIEW

**Comment by ingwarr** [comment]
It fails on destroy stage due to shared object used by both projects:

Step #4 - "destroy": 
Step #4 - "destroy": Error: Error reading Router: googleapi: Error 400: The router resource 'projects/ci-vpn-e2ed/regions/us-central1/routers/cr-uscentral1-to-prod-vpc-tunnels' is already being used by 'projects/ci-vpn-e2ed/regions/us-central1/vpnTunnels/vpn-tn-us-ce1-mgt-prd-internal-1', resourceInUseByAnotherResource
Step #4 - "destroy": 
Step #4 - "destroy": 
Step #4 - "destroy": 
Step #4 - "destroy": Error: Error reading Router: googleapi: Error 400: The router resource 'projects/ci-vpn-a85d/regions/us-central1/routers/cr-uscentral1-to-mgt-vpc-tunnels' is already being used by 'projects/ci-vpn-a85d/regions/us-central1/vpnTunnels/vpn-tn-us-ce1-prd-mgt-internal-1', resourceInUseByAnotherResource
Step #4 - "destroy": 
Step #4 - "destroy": 
Step #4 - "destroy": >>>>>> ------Exception-------
Step #4 - "destroy": >>>>>> Class: Kitchen::ActionFailed
Step #4 - "destroy": >>>>>> Message: 1 actions failed.
Step #4 - "destroy": >>>>>> Destroy failed on instance <single-tunnels-terraform>. Please see .kitchen/logs/single-tunnels-terraform.log for more details
Step #4 - "destroy": >>>>>> ----------------------
Step #4 - "destroy": >>>>>> Please see .kitchen/logs/kitchen.log for more details
Step #4 - "destroy": >>>>>> Also try running `kitchen diagnose --all` for configuration
Finished Step #4 - "destroy"
ERROR
ERROR: build step 4 "gcr.io/cloud-foundation-cicd/cft/developer-tools:0.1.0" failed: exit status 20

File: test/setup/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+terraform.tfvars
```
[the remark this answers is not in the document]
Comment (ingwarr) [reply test/setup/.gitignore]: Copied from example, I think we don't need it in current case

File: test/setup/make_source.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# limitations under the License.
+
+echo "#!/usr/bin/env bash" > ../source.sh
+
+prod_project_id=$(terraform output prod_project_id)
+echo "export TF_VAR_prod_project_id='$prod_project_id'" >> ../source.sh
+mgt_project_id=$(terraform output mgt_project_id)
+echo "export TF_VAR_mgt_project_id='$mgt_project_id'" >> ../source.sh
+prod_network="default"
+echo "export TF_VAR_prod_network='$prod_network'" >> ../source.sh
```
[the remark this answers is not in the document]
Comment (ingwarr) [reply test/setup/make_source.sh]: Done, thank you

File: test/setup/make_source.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+mgt_network="default"
+echo "export TF_VAR_mgt_network='$mgt_network'" >> ../source.sh
```
[the remark this answers is not in the document]
Comment (ingwarr) [reply test/setup/make_source.sh]: Done

File: test/setup/variables.tf [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+variable "folder_id" {
+  description = "The folder to deploy in"
+}
+
+variable "billing_account" {
+  description = "The billing account id associated with the project, e.g. XXXXXX-YYYYYY-ZZZZZZ"
+}
+
+variable "auto_create_network" {
```
Comment (aaron-lane) [inline test/setup/variables.tf]: This variable should be removed because it is not used.

**Comment by aaron-lane** [comment]
@ingwarr there is a simple fix for the destroy issue. Change the values of the `cr_name` arguments from strings to the `name` output of the associated `google_compute_router` resources.

File: test/setup/variables.tf (hunk 4 above)
Comment (ingwarr) [reply test/setup/variables.tf]: Ok, removed

**Comment by kopachevsky** [comment]
@aaron-lane can you please create build trigger for this one

**Comment by kopachevsky** [comment]
@morgante @aaron-lane as long it's existing issue, I've created separate issue for it, also please take a look on quick workaround (I know it's might be out of scope but it took me 20min to implement) https://github.com/terraform-google-modules/terraform-google-vpn/pull/24

**Comment by morgante** [comment]
@kopachevsky I don't think we should workaround by deleting the projects. The fact that `terraform destroy` cannot run properly is a bug which we should fix and implies incorrect dependency ordering in this repo.

**Comment by kopachevsky** [comment]
> @kopachevsky I don't think we should workaround by deleting the projects. The fact that `terraform destroy` cannot run properly is a bug which we should fix and implies incorrect dependency ordering in this repo.
 canceling PR

**Comment by aaron-lane** [comment]
@kopachevsky I identified a fix for the destroy issue. https://github.com/terraform-google-modules/terraform-google-vpn/pull/20#issuecomment-536095282

**Comment by kopachevsky** [comment]
> @kopachevsky I identified a fix for the destroy issue. [#20 (comment)](https://github.com/terraform-google-modules/terraform-google-vpn/pull/20#issuecomment-536095282)

checking

**Comment by kopachevsky** [comment]
@aaron-lane still same error

**Pull request merged**